core/sessions/ldapauth/ldap.go

```diff
@@ -0,0 +1,861 @@
+package ldapauth
+
+/*
+
+The LDAP authentication module forwards the credentials in the user session request
+for authentication with a configured upstream LDAP server
+
+This module relies on the two following local database tables:
+	ldap_sessions: 	Upon successful LDAP response, creates a keyed local copy of the user
+					email
+	ldap_user_api_tokens: User created API tokens, tied to the node, storing user email.
+						  Note: user can have only one API token at a time, and token
+						  expiration is enforced
+
+User session and roles are cached and revalidated with the upstream service at the interval defined in
+the local LDAP config through the Application.sessionReaper implementation in reaper.go.
+Changes to the upstream identity server will propogate through and update local tables (web sessions, API tokens)
+by either removing the entries or updating the roles. This sync happens for every auth endpoint hit, and
+via the defined sync interval. One goroutine is created to coordinate the sync timing in the New function
+
+This implementation is read only; user mutation actions such as Delete are not supported.
+
+MFA is supported via the remote LDAP server implementation. Sufficient request time out should accomodate
+for a blocking auth call while the user responds to a potential push notification callback.
+
+*/
+
+import (
+	"crypto/subtle"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-ldap/ldap/v3"
+	"github.com/pkg/errors"
```

Comment:
`github.com/pkg/errors` is now obsolete since the std lib has expanded support for various things:
```suggestion
	"errors"
```
The only case left for `github.com/pkg/errors` is full stack traces - if you are just wrapping/chaining errors then `fmt.Errorf("%w")` is preferred.

Reply:
Ah okay gotcha, I've updated all 9 occurrences. In sync.go as well, PTAL